### ISSUE TITLE
Multi-monitor support

### ISSUE THREAD

**Issue by bryphe** [onivim/oni#67]
A scenario that frequently occurs in a multiple-monitor setup at work is that I'd like to expand VIM to encompass multiple displays, and use the goodness of the easy window splitting / window navigation across monitors. Gvim on windows, at least, doesn't seem to handle this very well - I imagine the terminal clients might be better if the terminal emulator handles resize/multiple monitors, but in my case, it doesn't fit my workflow.

With Electron, it's relatively easy to change the BrowserWindow dimensions to encompass multiple screens (or alternatively, create multiple BrowserWindows that can communicate with other).  

It's easy to get the information about all displays:
```
    require("electron').screen.getAllDisplays()
```

And it's also pretty easy to size the browser window:
```
        require('electron').remote.getCurrentWindow().setPosition(x, y)
        require("electron").remote.getCurrentWindow().setSize(.., ..)
```

What would be helpful in terms of workflow would be a command that is like "MonitorExpandLeft, MonitorExpandRight, MonitorExpandDown, MonitorExpandUp". This would size the window to expand to the boundary of the window if it hasn't already, or expand it out to the next window. This could be bound to something like <C-W> plus arrow keys by default.

This would be somewhat gated by #20 since we need a nice way to express those commands, but we could have another hacky input added until we get that fixed..

**Comment by bryphe** [comment]
This is a feature I would love to have. This should allow for navigation between opened Oni windows, using `<C-w>l`, etc. The `WindowManager` can already handle and detect this case - the challenge here is bubbling that up, finding the next Oni window, and focusing it. The `main` process knows about all the active windows, so it'd be natural to send an `ipc` call to it to let it handle.

Testing with multiple Oni processes will be non-trivial here, but if we can have unit tests covering the navigation cases (figuring out which window is 'left', 'right', etc), that is sufficient.

**Comment by CrossR** [comment]
The bulk of this was fixed out in #1747. I'm going to make a second issue to track the remaining items I mentioned in the end of my PR.

**Comment by bryphe** [comment]
@CrossR - thanks for the fix here! It's awesome to have this in. Don't forget to claim the bounty on it! 👍 (Either bountysource or opencollective - whatever you prefer)
